src/components/Button/index.js

```diff
@@ -213,24 +213,27 @@ class Button extends Component {
             </Text>
         );
 
-        if (this.props.icon) {
+        if (this.props.icon || this.props.shouldShowRightIcon) {
             return (
                 <View style={[styles.justifyContentBetween, styles.flexRow]}>
                     <View style={[styles.alignItemsCenter, styles.flexRow, styles.flexShrink1]}>
-                        <View style={[styles.mr1, ...this.props.iconStyles]}>
-                            <Icon
-                                src={this.props.icon}
-                                fill={this.props.iconFill}
-                                small={this.props.small}
-                            />
-                        </View>
+                        {this.props.icon && (
+                            <View style={[styles.mr1, ...this.props.iconStyles]}>
+                                <Icon
+                                    src={this.props.icon}
+                                    fill={this.props.iconFill}
+                                    small={this.props.small}
+                                />
+                            </View>
+                        )}
                         {textComponent}
                     </View>
                     {this.props.shouldShowRightIcon && (
-                        <View style={styles.justifyContentCenter}>
+                        <View style={[styles.justifyContentCenter, styles.ml1, ...this.props.iconStyles]}>
```

Comment:
Looks like this has caused issues in such button components


<img width="1003" alt="Screenshot 2023-05-31 at 1 02 54 PM" src="https://github.com/Expensify/App/assets/32012005/0fba6520-588b-4b5b-9ad6-12fc1b49c35c">